### PR TITLE
#164829367 functionality for liking or disliking article

### PR DESCRIPTION
**Pull request by ipaullly**
#### Description
This PR edits and adds functionality in the ‘articles’ app. These changes allow authenticated users to ‘like’ or ‘dislike’ an article. Changes to the ‘ArticleSerializer’ will now ensure that articles are fetched from the database with the number of likes and dislikes. The following endpoints were implemented as specified in the 'Users should be able to like and or dislike articles' feature:

|  Method  | Endpoint |  Description |
|---|---|---|
| POST  | articles/:slug/like/ | Likes a particular article |
| POST  | articles/:slug/dislike/ | Dislikes a particular article |

#### Description of Task to be completed?
An authenticated user should be able to like or dislike articles written by others as well as those crafted by the user

*NOTE*
this feature requires you to sign-up and login before you can like or dislike articles on the site.

#### How should this be manually tested?
```
git clone --single-branch --branch ft-article-like-dislike-164829367 https://github.com/andela/ah-the-answer-backend.git
```
After cloning the branch, move into the generated project folder;
```
cd ah-the-answer-backend
```
Setup environment;
```
virtualenv venv
source venv/bin/activate
```
Install dependencies;
```
pip install -r requirements.txt
```
Create .env file and ensure all the environmental variables are set
Deactivate virtual environment and activate it again;
```
deactivate
source .env
```
Make migrations;
```
python manage.py makemigrations
python manage.py migrate
```
Run the server;
```
python manage.py runserver
```
Send a GET request to retrieve all articles and have access to the slugs. If there are no articles create one via the endpoint;
```
 POST-http://127.0.0.1:8000/api/articles/
```
Pass in the JSON body with the article details and ensure there is an authenticated token in the header;
```
{
	"article": {
		"title": "The Red Moon",
		"description": "A monologue on the dying of the stars",
		"body": "Isn't it beautiful what the cosmos beholds, don't you feel gleeful and at awe when you cast your eyes at starry nights. The stellar age is but the pre-adolescence of the Universe, and the evolution of life on earth but a second in the great chase to the end of time. the sun will shrink, then grow red and fiery, and with its rise, the planets will burn"
	} 
}
``` 
Once it is created, the response body will in include a 'slug' field
Test the like/dislike article APIs using POSTMAN. the request should contain the bearer header with a valid token and the 'slug' for a particular article is passed through the URL, i.e. 
```
 http://127.0.0.1:8000/api/articles/the-red-moon-df9c1ec9da6f/like/
```
And body:
Run tests with the following command:
```
python manage.py test
```

#### What are the relevant pivotal tracker stories?
#164829376

### CODE REVIEW

**Comment by WNjihia** [comment]
@ipaullly there are merge conflicts on your pr

**Review by MainaKamau92 — DISMISSED** [review]
I have tested the functionality locally and everything works great.

**Review by kihuha — DISMISSED** [review]

**Review by Raywire — DISMISSED** [review]
I like how the functionality works when you like and change your mind to dislike it reverses the like to -1 and the dislike to +1

**Comment by kihuha** [comment]
I would have however preferred to see the payload containing the updated data returned in this response in addition to returning the success message


Screenshot 2019-04-11 at 14 29 16

**Comment by ipaullly** [comment]
Now a user sees the article in the response body when they revert a like or dislike

**Review by Raywire — DISMISSED** [review]
The changes are welcomed.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -22,6 +25,14 @@ def find_article(slug):
             "message": "The article requested does not exist"
         })
 
+def find_article_title(slug):
```
Comment (WNjihia) [inline authors/apps/articles/views.py]: Hey I'm curious as to why you opted to define new find_article_title. Wouldn't it be easier if you referenced article.title?

**Review by Kyppy — DISMISSED** [review]
I liked the way you implemented your code to flip-flop the likes and dislikes. Your 'if' statements in the 'like_article' and 'dislike_article' methods could use some restructuring for readability. Also, you should include some doc strings briefly describing how it works because it is a bit confusing. Otherwise,  it is a cool feature.

**Review by Raywire — DISMISSED** [review]
By far the coolest feature. I have tested it locally and it works very well.
Screenshot 2019-04-11 at 20 45 31

**Comment by ipaullly** [comment]
You will now find in-line comments in the `LikeArticles`. I hope they provide some more insight into how the checks work and how ORM sends queries to the database within this functionality
.

**Review by LarryKarani — DISMISSED** [review]
I have tested it locally works fine. 
Screenshot 2019-04-11 at 21 01 04

**Review by kirega — APPROVED** [review]
Looks good

**Review by LarryKarani — APPROVED** [review]
Everything looks good locally

**Review by kihuha — APPROVED** [review]

**Comment by kihuha** [comment]
Functionality checks out

**Pull request merged**